Merge Conflict/Assets/Scripts/Upgrades/Upgrade.cs

```diff
@@ -35,17 +35,20 @@ public virtual string GetNextLevelDescription()
         return "";
     }
 
-    public int BuyNextLevel(int currentMoney)
+    public void BuyNextLevel()
     {
-        if(CostForNextLevel[Level] > currentMoney || isAtMaxLevel())
+        int currentMoney = MoneyHandler.Instance.Money;
+
+        if (CostForNextLevel[Level] > currentMoney || isAtMaxLevel())
```

Comment:
currently trhows an IndexOutOfBound Exception when upgrade is max level. 

Restructure if condition to "if (isAtMaxLevel() || CostForNextLevel[Level] > currentMoney)"

see https://stackoverflow.com/a/16711092